iothealth/device_health.py

```diff
@@ -77,8 +77,10 @@ def cameras(cls) -> Dict:
     @classmethod
     def _current_device(cls) -> _base_health.BaseHealth:
         if cls._current_device_cache is None:
-            if raspberry_pi.RaspberryPi().device_platform():
+            if "Raspberry Pi" in raspberry_pi.RaspberryPi().device_platform():
```

Comment:
Checking a specific string is a little bit error-prone. I think the better way is to check if  `raspberry_pi.RaspberryPi().device_platform()` return `None`. The function should return `None` if it is not a raspberry pi. i.e., in line 38 of `raspberry_pi.py`, it should return `None` not `str()`.

Reply:
I have made the change as suggested but ran into a problem that on the Jetson, the file on /sys/firmware/devicetree/base/model is not empty; it contains the string "NVIDIA Jetson Nano Developer Kit", and as a result, stderr will not be triggered causing the flow to enter the raspberry pi branch.

Any ideas on how to resolve this?

Reply:
> 
> 
> I have changed raspberry pi to inherent from Linux class. Since I don't have a raspberry pi with me at the moment, would need your help to perform a test to see if it still works or did I break something.
> 
> > Since the functionalities for Jetson Nano are very similar to Linux, I think it's a good time to refactor IoT Health. I would suggest we do this way:
> > 
> > * `linux.py` implements the features for the general Linux platform.
> > * `raspberry_pi.py` derives from `linux.py` and override the functions that are different from the Linux implementations such as `device_platform`, `temperature`, and `cameras`

Sure. I will test it and get back to you after the test. 👍

Reply:
Let me think about it 🤔

Reply:
A thought just occurred to me that since  /sys/firmware/devicetree/base/model is a system file, the string in the file should also be a fixed standard for every official release of the OS (still, it will probably change with every major release).

Therefore, it still might make sense to check the string for the device type. Probably try for an exact match instead of a substring check.

Reply:
Hmm... I guess using string (i.e., the name of the device) probably is the easiest way for now.

Reply:
In that case, could you give me the string in the Raspberry Pi? 

An exact match is probably going to be difficult for now, as I see people reporting different "Rev" number at the end of the string; i.e.; both "Raspberry Pi 3 Model B Rev 1.2" and "Raspberry Pi 3 Model B Rev 1.1" have been observed, so check for "Raspberry Pi 3 Model B" in the string?

Reply:
Probably still keep just checking "Raspberry Pi" for the time being, as it might also work for Raspberry Pi 4.